tools/metricsdocs/metricsdocs.go

```diff
@@ -0,0 +1,203 @@
+	"flag"
+	"fmt"
+	"log"
+	"net/http"
+	"os"
+	"strings"
+
+	"github.com/joho/godotenv"
+)
+
+func main() {
+	r := parseArguments()
+	r.checkoutProjects()
+	r.createDoc()
+}
+
+func parseArguments() *releaseData {
+	cacheDir := flag.String("cache-dir", "/tmp/release-tool", "The base directory used to cache git repos in")
+	configFile := flag.String("config-file", "", "Name of file containing the components versions")
+
+	flag.Parse()
+
+	if *configFile == "" {
+		log.Fatal("--config-file is a required argument")
+	}
+
+	org := "kubevirt"
+	baseDir := fmt.Sprintf("%s/%s/", *cacheDir, org)
+
+	return &releaseData{
+		org:      org,
+		projects: createProjects(*configFile, baseDir, org),
+	}
+}
+
+func createProjects(configFile string, baseDir string, org string) []*project {
+	config := getConfig(configFile)
+
+	var projects []*project
+	for _, n := range projectsInfo {
+		version, ok := config[n.short+"_VERSION"]
+
+		if !ok {
+			log.Fatalf("ERROR config doesn't contain '%s_VERSION' for %s", n.short, n.name)
+		}
+
+		projects = append(projects, &project{
+			short:   n.short,
+			name:    n.name,
+			version: version,
+
+			repoDir:        baseDir + n.name,
+			repoUrl:        fmt.Sprintf("https://github.com/%s/%s.git", org, n.name),
+			metricsDocPath: n.metricsDocPath,
+		})
+	}
+
+	return projects
+}
+
+func getConfig(configFile string) map[string]string {
+	config, err := godotenv.Read(configFile)
+	if err != nil {
+		log.Fatalf("ERROR reading %s config file", configFile)
+	}
+
+	return config
+}
+
+func (r *releaseData) checkoutProjects() {
+	for _, p := range r.projects {
+		err := p.gitCheckoutUpstream()
+		if err != nil {
+			log.Fatalf("ERROR checking out upstream: %s\n", err)
+		}
+		err = p.gitSwitchToBranch(p.version)
+		if err != nil {
+			log.Fatalf("ERROR changing to version branch: %s\n", err)
+		}
+	}
+}
+
+func (r *releaseData) createDoc() {
+	r.outFile = createFile()
+	defer r.outFile.Close()
+
+	r.writeHeader()
+
+	metrics := r.parseMetrics()
+
+	r.writeTOC(metrics)
+	r.writeMetrics(metrics)
+}
+
+func createFile() *os.File {
+	file, err := os.Create("docs/metrics.md")
+	if err != nil {
+		log.Fatalf("ERROR creating output file: %s", err)
+	}
+	return file
+}
+
+func (r *releaseData) writeHeader() {
+	r.outFile.WriteString("# KubeVirt components metrics\n\n")
+	r.outFile.WriteString("This document aims to help users that are not familiar with metrics exposed by all the KubeVirt components.\n")
+	r.outFile.WriteString("All metrics documented here are auto-generated in each component repository and gathered here.\n")
+	r.outFile.WriteString("They reflect and describe exactly what is being exposed.\n\n")
+}
+
+func (r *releaseData) parseMetrics() map[string]string {
+	metrics := make(map[string]string)
+
+	for _, p := range r.projects {
+		content, err := readLines(p.repoDir + "/" + p.metricsDocPath)
+		if err == nil {
+			parsed := p.parseMetricsDoc(content)
+			if len(parsed) != 0 {
+				metrics[p.name] = parsed
+			}
+		}
```

Comment:
Maybe worth to log an error message in case an error occurred while reading the md file. (unlikely, but it's better treat also the case that `err` is not `nil`.

Reply:
thanks